src/vs/workbench/contrib/chat/browser/actions/promptActions/dialogs/askToSelectPrompt/promptFilePickers.ts

```diff
@@ -140,8 +140,10 @@ export class PromptFilePickers {
 		const fileOptions = this._createPromptPickItems(options);
 		fileOptions.splice(0, 0, NEW_INSTRUCTIONS_FILE_OPTION);
 
+		const activeItem = options.resource && fileOptions.find(f => extUri.isEqual(f.value, options.resource));
```

Comment:
This does not fully fix the issue - the current file could be not in the instructions file list returned by the file locator, so in that case we want to add it to the list/preselect.

Reply:
I tried this, and it already works as you describe
- create an `foo.instructions.md` file in the workspace folder (not `.github/instructions`)
- open it in the editor and run the `Attach Instructions` command
- picker opens, 'foo' is selected